src/main/java/com/google/api/generator/engine/ast/TypeNode.java

```diff
@@ -97,6 +118,15 @@ public static boolean isReferenceType(TypeNode type) {
         && type.reference() != null;
   }
 
+  public static boolean isBoxedTypeEquals(TypeNode type1, TypeNode type2) {
+    if (type2.isPrimitiveType()) {
+      TypeNode type = type1;
+      type1 = type2;
+      type2 = type;
```

Comment:
Sorry, I am little confused here. Why do we need to do a swap here?

Reply:
the swap ensures that type1 is primitive and type2 is boxed type, then we do not have to worry about the order when passing in the arguments. For example, this method can be used to check `isBoxedTypeEquals(int, Integer)` or `isBoxedTypeEquals(Integer, int)`. Thanks

Reply:
done.